lib/statsd/instrument/prometheus/aggregators/timing.rb

```diff
@@ -4,15 +4,108 @@ module StatsD
   module Instrument
     module Prometheus
       module Aggregators
+        # Borrowed from: https://github.com/statsd/statsd/blob/master/lib/process_metrics.js#L22
+        # Read more about the format here: https://github.com/statsd/statsd/blob/master/docs/metric_types.md#timing
         class Timing < Base
           def aggregate
-            # TODO
-            datagrams.last
+            current_timer_data = { }
+            current_timer_count_data = { }
+            values = datagrams.map(&:value).sort
+            count = values.length
+            min = values.first
+            max = values.last
+
+            cumulative_values = [min]
+            cumulative_sum_squares_values = [min * min]
+            for i in 1..(count - 1) do
+              cumulative_values.push(values[i] + cumulative_values[i-1])
+              cumulative_sum_squares_values.push((values[i] * values[i]) + cumulative_sum_squares_values[i - 1])
+            end
+
+            sum = min
+            sum_squares = min * min
+            mean = min
+            threshold_boundary = max
+
+            percentiles.each do |percentile_threshold|
+              count_within_percentile_threshold = count.to_f
+
+              if (count > 1)
+                count_within_percentile_threshold = (percentile_threshold.abs / 100.0 * count).round
+                next if count_within_percentile_threshold == 0
+
+                if (percentile_threshold > 0)
+                  threshold_boundary = values[count_within_percentile_threshold - 1]
+                  sum = cumulative_values[count_within_percentile_threshold - 1]
+                  sum_squares = cumulative_sum_squares_values[count_within_percentile_threshold - 1]
+                else
+                  threshold_boundary = values[count - count_within_percentile_threshold]
+                  sum = cumulative_values[count - 1] - cumulative_values[count - count_within_percentile_threshold - 1]
```

Comment:
this line could be simpler as `cumulative_values.take(count_within_percentile_threshold)`, do we want to keep it as is for similarity with the existing ported code, or simplify it a little?